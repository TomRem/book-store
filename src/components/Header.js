import React, { Component } from 'react'

export class Header extends Component {
    state = {
        bookStoreName: "Black Moto",
        isCLicked: true,
        textColor: "white",
        backgroundColor: "black"
    };

    handleCLick = () => {

        if (this.state.isCLicked) {
            this.setState({
                textColor: "black",
                backgroundColor: "white",
                bookStoreName: "White Moto"
            })
        }
        else {
            this.setState({
                bookStoreName: "Black Moto",
                textColor: "white",
                backgroundColor: "black"
            })
        }
        this.setState({
            isCLicked: !this.state.isCLicked,
        })
    }

    render() {

        let headerCss = {
            color: this.state.textColor,
            backgroundColor: this.state.backgroundColor
        }

        return (
            <div className="row header" >
                <div className="col" style={headerCss} onClick={this.handleCLick}>
                    <h1>{this.state.bookStoreName}</h1>
                </div>
            </div>

        )
    }

}

export default Header
