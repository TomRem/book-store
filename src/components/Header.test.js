import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({
    adapter: new Adapter()
});

describe('Header tests', () => {

    it('Header renders without a problem', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Header changes text according to state', () => {
        const wrapper = shallow(<Header/>);
        // console.log(wrapper.debug());
        expect(wrapper.find('h1').text()).toBe('Black Moto');
        wrapper.setState({
            bookStoreName: "White Moto"
        })
        expect(wrapper.find('h1').text()).toBe('White Moto');
    })

    it('Header state changes after handleClick' , () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper.state().bookStoreName).toBe('Black Moto');
        wrapper.find('.col').simulate('click');
        expect(wrapper.state().bookStoreName).toBe('White Moto');
    })

    it('Snapshot matches', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    })
})