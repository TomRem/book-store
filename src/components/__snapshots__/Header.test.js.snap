// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Snapshot matches 1`] = `ShallowWrapper {}`;
